homeassistant/components/remote/harmony.py

```diff
@@ -184,113 +192,188 @@ def name(self):
+                              self.name,
+                              activity)
         else:
-            _LOGGER.error("No activity specified with turn_on service")
+            _LOGGER.error("%s: No activity specified with turn_on service",
+                          self.name)
 
     async def async_turn_off(self, **kwargs):
         """Start the PowerOff activity."""
-        await self._client.power_off()
+        import aioharmony.exceptions as aioexc
+        _LOGGER.debug("%s: Turn Off", self.name)
+        try:
+            await self._client.power_off()
+        except aioexc.TimeOut:
+            _LOGGER.error("%s: Powering off timed-out", self.name)
 
     # pylint: disable=arguments-differ
     async def async_send_command(self, command, **kwargs):
         """Send a list of commands to one device."""
+        from aioharmony.harmonyapi import SendCommandDevice
+        import aioharmony.exceptions as aioexc
+
+        _LOGGER.debug("%s: Send Command", self.name)
         device = kwargs.get(ATTR_DEVICE)
         if device is None:
-            _LOGGER.error("Missing required argument: device")
+            _LOGGER.error("%s: Missing required argument: device", self.name)
             return
 
         device_id = None
         if device.isdigit():
-            _LOGGER.debug("Device is numeric")
+            _LOGGER.debug("%s: Device %s is numeric",
+                          self.name, device)
             if self._client.get_device_name(int(device)):
                 device_id = device
 
-        if not device_id:
-            _LOGGER.debug("Find device ID based on device name")
-            device_id = self._client.get_activity_id(str(device).strip())
+        if device_id is None:
+            _LOGGER.debug("%s: Find device ID %s based on device name",
+                          self.name, device)
+            device_id = self._client.get_device_id(str(device).strip())
 
-        if not device_id:
-            _LOGGER.error("Device  %s is invalid", device)
+        if device_id is None:
+            _LOGGER.error("%s: Device %s is invalid", self.name, device)
             return
 
         num_repeats = kwargs.get(ATTR_NUM_REPEATS)
         delay_secs = kwargs.get(ATTR_DELAY_SECS, self._delay_secs)
 
+        # Creating list of commands to send.
+        snd_cmnd_list = []
         for _ in range(num_repeats):
             for single_command in command:
-                _LOGGER.debug("Sending command %s", single_command)
-                await self._client.send_command(device, single_command)
-                await asyncio.sleep(delay_secs)
+                send_command = SendCommandDevice(
+                    device=device,
+                    command=single_command,
+                    delay=0
+                )
+                snd_cmnd_list.append(send_command)
+                if delay_secs > 0:
+                    snd_cmnd_list.append(float(delay_secs))
```

Comment:
Does the client interpret floats as a delay?

Reply:
Yes, it does. Just a float means a delay. :-)